src/hypothesis/extra/django/models.py

```diff
@@ -137,26 +136,19 @@ def _get_strategy_for_field(f):
         if isinstance(f, (dm.CharField, dm.TextField)) and f.blank:
             choices.append(u'')
         strategy = st.sampled_from(choices)
-    elif isinstance(f, dm.EmailField):
-        return emails
+    elif isinstance(f, dm.SlugField):
+        strategy = st.text(alphabet=safe_letters,
+                           min_size=(None if f.blank else 1),
+                           max_size=f.max_length)
     elif type(f) in (dm.TextField, dm.CharField):
         strategy = st.text(min_size=(None if f.blank else 1),
                            max_size=f.max_length)
-    elif type(f) == dm.DecimalField:
-        m = 10 ** f.max_digits - 1
-        div = 10 ** f.decimal_places
-        q = Decimal('1.' + ('0' * f.decimal_places))
-        strategy = (
-            st.integers(min_value=-m, max_value=m)
-            .map(lambda n: (Decimal(n) / div).quantize(q)))
+    elif isinstance(f, dm.DecimalField):
+        bound = Decimal(10 ** f.max_digits - 1) / (10 ** f.decimal_places)
```

Comment:
Read it again - both versions are right.  We now get the integer with `max_digits` nines, then insert the decimal place; the old version is more convoluted but gets the same result.